src-tauri/src/api/api.rs

```diff
@@ -1,37 +1,71 @@
-            };
-            let payload = serde_json::to_string(&login_request).unwrap();
-            let uri = format!("{server_url}{API_VERSION}users/login");
-            info!("call: {uri} payload: {payload}");
-            let response = client
-                .post(uri)
-                .body(payload)
-                .send()
-                .await
-                .unwrap();
-            match &response.status() {
-                &StatusCode::OK => {
-                    let login_from_header: &str = &response.headers().get("token").unwrap().to_str().unwrap();
-                    let token: String = login_from_header.to_owned();
-                    let user: UserResponse = response.json::<UserResponse>().await.unwrap();
-                    Ok(Response::LoginResponse(token, user))
+        ApiEvent::LoginEvent(login_id, password) => login(client, create_uri(server_url, "users/login"), login_id, password).await,
+        ApiEvent::MyTeams => my_teams(client, create_uri(server_url, "users/me/teams"), token).await
+    }
+}
+
+fn create_uri(server_url: &str, endpoint: &str) -> String {
+    format!("{server_url}{API_VERSION}{endpoint}")
+}
+
+async fn handle<T: Serialize>(client: &Client, method: Method, url: String, payload: Option<T>, token: Option<&String>) -> reqwest::Response {
+    let mut builder = client.request(method, url);
+    builder = match payload {
+        Some(json) => builder.json(&json),
+        _ => builder
+    };
+    builder = match token {
+        Some(bearer_token) => builder.bearer_auth(format!("Bearer {bearer_token}")),
+        _ => builder
+    };
+    builder.send().await.unwrap()
+}
+
+async fn login(client: &Client, uri: String, login: &String, password: &String) -> Result<Response, Error> {
+    let login_request = LoginRequest {
+        login_id: (&login).to_string(),
+        password: (&password).to_string(),
+    };
+    let response = handle(client, Method::POST, uri, Some(login_request), None).await;
+    match &response.status() {
+        &StatusCode::OK => {
+            let token = get_token(&response.headers()).to_owned();
+            let user_response = &response.json::<UserResponse>().await;
+            match user_response {
+                Ok(user) => {
+                    let UserResponse { id, username, .. } = user;
+                    Ok(Response::LoginResponse(token, id.to_owned(), username.to_owned()))
                 }
-                _ => Err(Default::default()),
+                Err(_) => Err(Default::default()),
             }
         }
+        _ => Err(Default::default()),
+    }
+}
+
+fn get_token(headers: &HeaderMap) -> &str {
+    let token = headers.get("token").unwrap();
+    token.to_str().unwrap()
+}
+
+async fn my_teams(client: &Client, uri: String, token: Option<&String>) -> Result<Response, Error> {
+    let response = handle(client, Method::GET, uri, None as Option<()>, token).await;
+    match &response.status() {
+        &StatusCode::OK => {
```

Comment:
Status is an u16 and implement Copy trait. Pointer is u64 so you actually waisting here 3x memory

```suggestion
    match response.status() {
        StatusCode::OK => {
```